selfservice/strategy/password/login_test.go

```diff
@@ -677,6 +677,24 @@ func TestCompleteLogin(t *testing.T) {
 		assert.Equal(t, identifier, gjson.Get(body2, "identity.traits.subject").String(), "%s", body2)
 	})
 
+	t.Run("should login same identity regardless of leading or trailing whitespace", func(t *testing.T) {
+		identifier, pwd := x.NewUUID().String(), "password"
+		createIdentity(identifier, pwd)
+
+		browserClient := testhelpers.NewClientWithCookies(t)
+		f := testhelpers.InitializeLoginFlowViaBrowser(t, browserClient, publicTS, false, false)
+
+		values := url.Values{"method": {"password"}, "password_identifier": {"  " + identifier + "  "}, "password": {pwd}, "csrf_token": {x.FakeCSRFToken}}.Encode()
+
+		_, res := testhelpers.LoginMakeRequest(t, false, false, f, browserClient, values)
+		assert.EqualValues(t, http.StatusOK, res.StatusCode)
+
+		f = testhelpers.InitializeLoginFlowViaBrowser(t, browserClient, publicTS, true, false)
+		body2, res := testhelpers.LoginMakeRequest(t, false, false, f, browserClient, values)
```

Comment:
So you are making 2 requests here, but don't really use the first one? I mean, the values are the same in both requests.

Reply:
You are right, that is a bit weird. Honestly I just copy-pasted the capitalization test case from above and changed the identifier transformation. I think the capitalization test case might originate from `"should be a new session with refresh flag"` which would explain why there are two requests.

I can push a simplified version of both my new test case and the one above (as it suffers from the same problem), if you like?

Reply:
Yeah, makes sense.